src/main/java/org/skriptlang/skript/bukkit/recipes/elements/SecRegisterRecipe.java

```diff
@@ -0,0 +1,272 @@
+@Description({
+	"Create a custom recipe for any of the following types:",
+	"Shaped, Shapeless, Blasting, Furnace, Campfire, Smoking, Smithing Transform, Smithing Trim or Stonecutting",
+	"NOTES:",
+	"All recipes except Smithing Trim require a 'result item'",
+	"Shaped and Shapeless Ingredients allows custom items only on Paper",
+	"Shaped and Shapeless have a maximum of 9 and minimum requirement of 2 ingredients",
+	"Blasting, Furnace, Campfire and Smoking all fall under Cooking Recipe Type",
+	"Groups only apply to Shaped, Shapeless and Cooking Recipes",
+	"Category only applies to Shaped, Shapeless and Cooking Recipes",
+	"You can not create a Cooking, Crafting and Complex Recipe type."
+})
+@Examples({
+	"register a new shaped recipe with the name \"my_recipe\":",
+		"\tset the recipe ingredients to diamond, air, diamond, air, emerald, air, diamond and diamond #OR",
+		"\tset recipe ingredients of 1st row to diamond, air and diamond",
+		"\tset recipe ingredients of second row to air, emerald and air",
+		"\tset recipe ingredients of 3rd row to diamond, air and diamond",
+		"\tset recipe group to \"my group\"",
+		"\tset recipe crafting category to crafting misc",
+		"\tset recipe result to diamond sword named \"Heavenly Sword\"",
+	"",
+	"register a shapeless recipe with namespace \"my_recipe\":",
+		"\tset recipe ingredients to 3 diamonds, 3 emeralds and 3 iron ingots",
+		"\tset the recipe group to \"custom group\"",
+		"\tset the recipe crafting category to crafting category misc",
+		"\tset the recipe result item to diamond helmet named \"Heavenly Helm\"",
+	"",
+	"#Furnace, Campfire and Smoking follow same format as Blasting example",
+	"create new blasting recipe with the namespacekey \"my_recipe\":",
+		"\tset the recipe experience to 5",
+		"\tset the recipe cooking time to 10 seconds",
+		"\tset the recipe group to \"custom group\"",
+		"\tset the recipe cooking category to cooking misc",
+		"\tset the recipe input item to coal named \"Ash\"",
+		"\tset the recipe resulting item to gunpowder named \"Dust\"",
+	"",
+	"#Smithing Trim follows the same format, except for 'result item'",
+	"create smithing transform recipe with key \"my_recipe\":",
+		"\tset the recipe base item to diamond helmet",
+		"\tset the recipe template item to paper named \"Blueprint\"",
+		"\tset the recipe addition item to netherite ingot named \"Pure Netherite\"",
+		"\tset the recipe result to netherite helmet named \"Pure Helmet\"",
+	"",
+	"create a new stonecutting recipe with namespacekey \"my_recipe\":",
+		"\tset the recipe source item to cobblestone named \"Cracked Stone\"",
+		"\tset the recipe result to stone named \"Refurnished Stone\""
+})
+@Since("INSERT VERSION")
+public class SecRegisterRecipe extends Section {
+
+	private static final boolean RUNNING_1_20 = Skript.isRunningMinecraft(1, 20, 0);
+
+	static {
+		Skript.registerSection(SecRegisterRecipe.class, "(register|create) [a] [new] %*recipetype% with [the] (key|id) %string%");
```

Comment:
What if I want to create a recipe without registering it? I think that this should wait for secspressions and utilise that feature instead. An expression section to create the recipe, and an effect to register a recipe (could just be the server recipe's expression changer i mentioned before (could support set, add, remove, reset, clear, etc))

Reply:
Is your idea behind it just this?? Since if so I'd agree it's nice, but I think it could be added as a future thing unless it's confirmed Secpressions are 2.10
```applescript
set {_recipeCopy} to a new %recipetype% with [the] (id|key) %string%:
    set recipe cook time to 10 seconds
    return recipe#???
```

Reply:
```applescript
set {_recipeCopy} to a new %recipetype% with [the] (id|key) %string%:
    set recipe cook time to 10 seconds
``` just this

Reply:
And they are feature-ready, but just waiting on a double check by walrus